webapi/src/main/scala/org/knora/webapi/messages/v2/responder/valuemessages/ValueMessagesV2.scala

```diff
@@ -2820,11 +2820,65 @@ object HierarchicalListValueContentV2 extends ValueContentReaderV2[HierarchicalL
       throw BadRequestException(s"List node IRI <$listValueAsListNode> is not a Knora data IRI")
     }
 
-    HierarchicalListValueContentV2(
-      ontologySchema = ApiV2Complex,
-      valueHasListNode = listValueAsListNode.toString,
-      comment = getComment(jsonLDObject)
-    )
+    for {
+      isRootNode <- checkIfNodeIsRoot(
```

Comment:
This check seems to be request specific. Did you check if `fromJsonLDObject` should never allow a root node?

Reply:
I don't get your question. In the case of list values, yes it shouldn't be allowed a root node as value. That's actually the clue of this bug.

Reply:
I'm not sure, that this check is implemented in the correct place. It should be probably better to move it to here: https://github.com/dasch-swiss/dsp-api/blob/3a0902ea9b44860bcb73e4ef25888f5ac8a5dc21/webapi/src/main/scala/org/knora/webapi/responders/v2/ValuesResponderV2.scala#L195-L201

This way, you will also be able to write the missing tests.

Reply:
@subotic Good point, moved checking method to `ResourceUtilV2` and added test to `ValuesResponderV2Spec`.